pkg/scheduler/controller/shoot/reconciler.go

```diff
@@ -226,31 +226,21 @@ func filterSeedsMatchingMultiZonalAttribute(seedList []gardencorev1beta1.Seed, s
 	if !gardenerschedulerfeatures.FeatureGate.Enabled(features.HAControlPlanes) {
 		return seedList, nil
 	}
-
-	var (
-		singleZonalSeeds []gardencorev1beta1.Seed
-		multiZonalSeeds  []gardencorev1beta1.Seed
-	)
-
-	for _, seed := range seedList {
-		if metav1.HasLabel(seed.ObjectMeta, v1beta1constants.LabelSeedMultiZonal) {
-			multiZonalSeeds = append(multiZonalSeeds, seed)
-			continue
+	// only multi-zonal seeds can be candidates for shoots whose control plane has failure tolerance = zone
```

Comment:
Does it make sense to adapt https://github.com/gardener/gardener/tree/master/test/integration/scheduler/shoot?

Reply:
Yes it makes sense. There are other changes that are also coming to scheduler as part of GEP-20. Can we add additional integration tests as part of other PR (yet to be filed)?

Reply:
It's hard for me to keep track of the overall story (already now there are quite some open PRs and back-and-forth). If we postpone it now it will be difficult to ensure that respective tests were indeed added. Given that the changes in this PR w.r.t. scheduler are not too large, I would prefer to add it now so that the entire PR contains "everything that is needed".

Reply:
Added